rita_common/src/network_monitor/mod.rs

```diff
@@ -102,84 +83,62 @@ pub struct IfaceStats {
+        if let Some(packet_loss_stats) = (*(NETWORK_MONITOR.write().unwrap())).packet_loss_history.get(iface) {
+            stats.insert(
+                iface.clone(),
+                IfaceStats {
+                    latency: LatencyStats {
+                        avg: latency_stats.get_avg(),
+                        std_dev: latency_stats.get_std_dev(),
                     },
-                );
-            } else {
-                error!("Found entry in one that's not in the other ")
-            }
+                    packet_loss: PacketLossStats {
+                        avg: packet_loss_stats.get_avg(),
+                        five_min_avg: packet_loss_stats.get_five_min_average(),
+                    },
+                },
+            );
+        } else {
+            error!("Found entry in one that's not in the other ")
         }
-
-        Ok(stats)
     }
+    stats
 }
 
 pub struct GetNetworkInfo;
 
-impl Message for GetNetworkInfo {
-    type Result = Result<NetworkInfo, RitaCommonError>;
-}
-
-impl Handler<GetNetworkInfo> for NetworkMonitor {
-    type Result = Result<NetworkInfo, RitaCommonError>;
-
-    fn handle(&mut self, _msg: GetNetworkInfo, _ctx: &mut Context<Self>) -> Self::Result {
-        match self.last_babel_dump.clone() {
-            Some(dump) => Ok(dump),
-            None => Err(RitaCommonError::MiscStringError(
-                "No babel info ready!".to_string(),
-            )),
-        }
+pub fn get_network_info(_msg: GetNetworkInfo) -> Result<NetworkInfo, RitaCommonError> {
+    match (*(NETWORK_MONITOR.write().unwrap())).last_babel_dump.clone() {
+        Some(dump) => Ok(dump),
+        None => Err(RitaCommonError::MiscStringError("No babel info ready!".to_string())),
     }
 }
 
-#[derive(Message, Clone)]
+#[derive(Clone)]
 pub struct NetworkInfo {
     pub babel_neighbors: Vec<BabelNeighborLegacy>,
     pub babel_routes: Vec<BabelRouteLegacy>,
     pub rita_neighbors: Vec<RitaNeighbor>,
 }
 
-impl Handler<NetworkInfo> for NetworkMonitor {
-    type Result = ();
-
-    fn handle(&mut self, msg: NetworkInfo, _ctx: &mut Context<Self>) -> Self::Result {
-        let babel_neighbors = &msg.babel_neighbors;
-        let babel_routes = &msg.babel_routes;
-        let rita_neighbors = &msg.rita_neighbors;
-        observe_network(
-            babel_neighbors,
-            rita_neighbors,
-            &mut self.latency_history,
-            &mut self.packet_loss_history,
-        );
-        network_stats(babel_routes, babel_neighbors);
-        self.last_babel_dump = Some(msg);
-    }
+/// updates babel neighbors, babel routes, and rita neighbors for a NetworkInfo
+pub fn update_network_info(msg: NetworkInfo) {
+    let babel_neighbors = &msg.babel_neighbors;
+    let babel_routes = &msg.babel_routes;
+    let rita_neighbors = &msg.rita_neighbors;
+    observe_network(
+        babel_neighbors,
+        rita_neighbors,
+        &mut (*(NETWORK_MONITOR.write().unwrap())).latency_history,
```

Comment:
good job figuring out how to deal with the references here, but as noted previously it's better to get one mutable object and pass it, that way we only need to get the lock once.

let mut network_monitor = NETWORK_MONITOR.write().unwrap();

this actaully frees you from doing a lot of reference operations becuase .latency_history will still 'just work' this is becuase the '.' statement in Rust can do a ton of lifting under the hood to remove referencing/dereferencing etc so long as the compiler can prove to itself that it's safe.